client/src/sections/dashboard/trainee/CoursesSection.tsx

```diff
@@ -1,39 +1,100 @@
+import { addCourse, resetCourseList } from '@/src/features/trainee/dashboard/traineeCourseSlice';
+import { useAppDispatch, useAppSelector } from '@/src/redux/hooks';
+import { useGetTraineeCoursesQuery } from '@/src/services/traineeAPI';
 import TraineeCourseCard from '@/src/shared/components/Card/TraineeCourseCard';
 import SortDropdown from '@/src/shared/components/Dropdown/SortDropdown/SortDropdown';
 import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import FilterIcon from '@/src/shared/icons/FilterIcon';
 import ShowIcon from '@/src/shared/icons/ShowIcon';
+import { type CourseCollection } from '@/src/shared/utils';
 import Link from 'next/link';
+import { Fragment, useEffect, useState } from 'react';
 
 const CoursesSection = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
```

Comment:
May I know why you are declaring this page again in the components, I mean you already have the page in the slice, which is by the way I do not see you using it